extensions/src/org/pentaho/platform/plugin/services/security/userrole/jdbc/JdbcUserRoleListService.java

```diff
@@ -241,6 +255,13 @@ protected Object mapRow( final ResultSet rs, final int rownum ) throws SQLExcept
       }
     }
 
+    // Now add extra role if it does not exist in the list
+    for ( String extraRole : extraRoles ) {
```

Comment:
In our case, every authenticated user get the "Authenticated" role. This does not exist in the third-party security provider. So, yes every authenticated user is expected to have atleast one extra role.